### ISSUE TITLE
Server Page console output locks program after very long time.

### ISSUE THREAD

**Issue by J0nathan550** [stayintarkov/SIT.Manager.Avalonia#57]
Not sure how to explain more brief then that. 
Played for 5 hours straight from the SIT.Manager.Avalonia with server turned on inside of the app.
When I finished playing, I wanted to turned off the server which then locked completely app (not crashed, just locked forever)
It did closed the server (I viewed the Task Manager) but after application is not responsive anymore (even Windows System doesn't tell me that this application does not response anymore). 

We have to look inside of Server Page log and see what can cause the issue, program doesn't crash, just locks itself without allowing to click on anything (or even close itself).

Footage:

https://cdn.discordapp.com/attachments/947494021360656434/1213649967995035768/2024-03-03_01-50-12.mp4?ex=65f63e9c&is=65e3c99c&hm=fab1b3f7828fa7b904a0eb54b588b6e6433054b25b859dcb99beef6a0d0a1975&

**Comment by J0nathan550** [comment]
> Not sure how to explain more brief then that. Played for 5 hours straight from the SIT.Manager.Avalonia with server turned on inside of the app. When I finished playing, I wanted to turned off the server which then locked completely app (not crashed, just locked forever) It did closed the server (I viewed the Task Manager) but after application is not responsive anymore (even Windows System doesn't tell me that this application does not response anymore).
> 
> We have to look inside of Server Page log and see what can cause the issue, program doesn't crash, just locks itself without allowing to click on anything (or even close itself).
> 
> Footage:
> 
> https://cdn.discordapp.com/attachments/947494021360656434/1213649967995035768/2024-03-03_01-50-12.mp4?ex=65f63e9c&is=65e3c99c&hm=fab1b3f7828fa7b904a0eb54b588b6e6433054b25b859dcb99beef6a0d0a1975&

That was pure publish build for windows using my fork.
https://github.com/J0nathan550/SIT.Manager.Avalonia

**Comment by Plootie** [comment]
I'll investigate this once I have linux sorted out

**Comment by Plootie** [comment]
Cannot repro. I didn't have time to play for hours but I wrote a program to spam the ping endpoint ~20,000 times to fill the console log to the maximum value and observed no hanging. It's possible this is caused by something more complicated than just the server log but for now I can't tell what that might be

**Comment by J0nathan550** [comment]
Cannot reproduce either, was maybe the rare case? 😄

**Comment by Plootie** [comment]
Hopefully? If this happens again or you figure out repro steps for this I'll reopen this issue and we'll find a fix for it

**Comment by J0nathan550** [comment]
Could repro once again...

https://github.com/stayintarkov/SIT.Manager.Avalonia/assets/88881607/87ffaa4b-9e30-4439-babe-20f176f51ca2

**Comment by Plootie** [comment]
Are you able to find repro steps at all? I don't have any idea what could be causing the manager to lock up in this fashion

**Comment by J0nathan550** [comment]
Yeah main thing is to play game for too long

**Comment by J0nathan550** [comment]
Also it may encouter the bug that the process of server is getting seperated from the launcher.

**Comment by Plootie** [comment]
Gonna reopen this as it's clearly not an isolated issue but until we have repro steps there's not a whole lot I can do about this

**Comment by artehe** [comment]
There's one thing which comes to mind which we could try to start with, adding this to the ProcessStartInfo ```csharp RedirectStandardError = !cal,``` just in case theres a load of data being sent to standard error as well as the standard out that we currently take data from. Causing it to behave in the same way that we had seen before if the server is able to be running but the launcher it was attached to closed.

If that fails then there is the possibility that we could start the window in the same was as we could if close after launch is on, but embed the window in the server page rather than let it pop out. This would mean that you wouldn't have the features like changing the console font and color but I don't think that would be the end of the world if it works reliably.

**Comment by J0nathan550** [comment]
The issue of deadlocking maybe in threads. Some of the task executing in event thread can deadlock itself. I will try to understand later. I had simillar issue in my SoundScapes program. But not about server, was about the switching songs in the application. And because I used tasks + event system of one library I got deadlock every single time. Had to do a workaround with timers and no more deadlocks.
